---
title: "RNA-seq Commands on MARCC"
date: "2018-04-13"
---

RSEM
SAMtools
Bowtie

```
ssh gateway2.marcc.jhu.edu -l username@jhu.edu

srun -p shared -c 1 --mem 1024 -t 30 --pty bash

module load anaconda-python/3.6
source activate pg2018

cd ~/scratch
# mkdir genomes
cd genomes
# cp ~/work/genomes/chr20.fa.gz .
# gunzip chr20.fa.gz
cp ~/work/genomes/hg19-iGenomes.gtf .
cp ~/work/genomes/hg19.sizes .

module load bowtie/1.2.2
rsem-prepare-reference --bowtie --gtf hg19-iGenomes.gtf chr20.fa chr20rsem

cd ..
cp -R ~/work/rawdata .
cd rawdata
./doRename.sh ALIAS
gunzip HUES64.mRNA*gz

cd ..
mkdir rna-seq
cd rna-seq

rsem-calculate-expression --output-genome-bam --paired-end --strandedness none ../rawdata/HUES64.mRNA.chr20.R1.fastq ../rawdata/HUES64.mRNA.chr20.R2.fastq ../genomes/chr20rsem HUES64_1

gunzip ~/scratch/rawdata/ectoderm.mRNA*gz
rsem-calculate-expression --output-genome-bam --paired-end --strandedness none ~/scratch/rawdata/ectoderm.mRNA.chr20.R1.fastq ~/scratch/rawdata/ectoderm.mRNA.chr20.R2.fastq ~/scratch/genomes/chr20rsem ectoderm_1

rsem-generate-data-matrix HUES64_1.genes.results ectoderm_1.genes.results > HUES64_vs_ectoderm.genes.matrix
rsem-run-ebseq HUES64_vs_ectoderm.genes.matrix 1,1 HUES64_vs_ectoderm.genes.EBSeq
rsem-control-fdr HUES64_vs_ectoderm.genes.EBSeq 0.05 HUES64_vs_ectoderm.genes.DiffExp_fdr.txt

samtools sort -o HUES64_1.genome_sorted.bam HUES64_1.genome.bam
samtools index HUES64_1.genome_sorted.bam
rsem-bam2wig HUES64_1.genome_sorted.bam HUES64_1.genome.wig HUES64_1
wigToBigWig HUES64_1.genome.wig ../genomes/hg19.sizes HUES64_1.genome.bw

samtools sort -o ectoderm_1.genome_sorted.bam ectoderm_1.genome.bam
samtools index ectoderm_1.genome_sorted.bam
rsem-bam2wig ectoderm_1.genome_sorted.bam ectoderm_1.genome.wig ectoderm_1
wigToBigWig ectoderm_1.genome.wig ../genomes/hg19.sizes ectoderm_1.genome.bw
```